<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TabGeometry</class>
 <widget class="QWidget" name="TabGeometry">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1110</width>
    <height>755</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Geometry</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QSplitter" name="splitter_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="childrenCollapsible">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="verticalLayoutWidget">
      <layout class="QGridLayout" name="gridLayout">
       <item row="4" column="0" colspan="2">
        <widget class="QListWidget" name="listMembranes">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="btnRemoveCompartment">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Remove</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="btnAddCompartment">
         <property name="text">
          <string>Add</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QListWidget" name="listCompartments">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="whatsThis">
          <string>&lt;p&gt;A list of all the compartments in the model.&lt;/p&gt;
&lt;p&gt;To assign a compartment to an area of the geometry image:
&lt;ul&gt;
&lt;li&gt;Click on the compartment name, then on the 'Select compartment geometry...' button (or simply double-click on the compartment name)&lt;/li&gt;
&lt;li&gt;Click on the corresponding point on the geometry image on the left&lt;/li&gt;
&lt;/ul&gt;
&lt;/p&gt;
&lt;p&gt;The Compartment settings on the right will be updated to show an image of the new shape of the compartment&lt;/p&gt;</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="lblCompartments">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Compartments:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="2">
        <widget class="QLabel" name="lblMembranes">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Membranes:</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="grpCompartment">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Compartment Settings</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QGridLayout" name="gridCompartmentSettings">
         <item row="0" column="1">
          <widget class="QLineEdit" name="txtCompartmentName">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>The size of the compartment</string>
           </property>
           <property name="statusTip">
            <string>The size of the compartment</string>
           </property>
           <property name="whatsThis">
            <string>The size of the compartment</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="lblCompColourLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <weight>50</weight>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>Colour:</string>
           </property>
           <property name="scaledContents">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="lblCompartmentNameLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>The size of the compartment</string>
           </property>
           <property name="statusTip">
            <string>The size of the compartment</string>
           </property>
           <property name="whatsThis">
            <string>The size of the compartment</string>
           </property>
           <property name="text">
            <string>Name:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QLabel" name="lblCompartmentColour">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <weight>50</weight>
             <bold>false</bold>
            </font>
           </property>
           <property name="toolTip">
            <string>The colour of the compartment</string>
           </property>
           <property name="statusTip">
            <string>The colour of the compartment</string>
           </property>
           <property name="whatsThis">
            <string>&lt;h4&gt;Compartment Colour&lt;/h4&gt;
&lt;p&gt;The compartment geometry is generated from all the pixels of this colour in the compartment geometry image on the left.&lt;/p&gt;
&lt;p&gt;To change the colour, click on 'Select compartment geometry...', then click on the geometry image on the left.&lt;/p&gt;</string>
           </property>
           <property name="autoFillBackground">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>none</string>
           </property>
           <property name="scaledContents">
            <bool>true</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QLabel" name="lblCompSize">
           <property name="text">
            <string>Volume/Area:</string>
           </property>
           <property name="textInteractionFlags">
            <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTabWidget" name="tabCompartmentGeometry">
         <property name="toolTip">
          <string>The boundaries of the geometry of the compartments</string>
         </property>
         <property name="statusTip">
          <string>The boundaries of the geometry of the compartments</string>
         </property>
         <property name="whatsThis">
          <string>The boundaries of the geometry of the compartments</string>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tabCompartmentImage">
          <attribute name="title">
           <string>Compartment Pixels</string>
          </attribute>
          <layout class="QHBoxLayout" name="horizontalLayout_13">
           <item>
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="0" column="0">
              <widget class="QLabelMouseTracker" name="lblCompShape" native="true">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>The shape of the compartment</string>
               </property>
               <property name="whatsThis">
                <string>&lt;h4&gt;Compartment Geometry&lt;/h4&gt;
&lt;p&gt;The compartment geometry is generated from all the pixels in the compartment geometry image on the left that have the colour assigned to this compartment&lt;/p&gt;
&lt;p&gt;To change the geometry, click 'Select compartment geometry...', then click on the geometry image on the left.&lt;/p&gt;</string>
               </property>
               <property name="text" stdset="0">
                <string>none</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="btnChangeCompartment">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string/>
               </property>
               <property name="statusTip">
                <string/>
               </property>
               <property name="whatsThis">
                <string>&lt;h4&gt;Select compartment geometry...&lt;/h4&gt;
&lt;p&gt;To assign a geometry to this compartment, click here, then click on the compartment geometry image on the left .&lt;/p&gt;
&lt;p&gt;The shape of the compartment will be generated from all the pixels in the image of the same colour as the one that was clicked on&lt;/p&gt;</string>
               </property>
               <property name="text">
                <string>Select compartment geometry...</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tabCompartmentBoundary">
          <attribute name="title">
           <string>Boundary Lines</string>
          </attribute>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="1" column="1">
              <widget class="QSpinBox" name="spinBoundaryIndex">
               <property name="wrapping">
                <bool>true</bool>
               </property>
               <property name="maximum">
                <number>0</number>
               </property>
              </widget>
             </item>
             <item row="0" column="0" colspan="7">
              <widget class="QScrollArea" name="scrollBoundaryLines">
               <property name="widgetResizable">
                <bool>true</bool>
               </property>
               <widget class="QWidget" name="scrollBoundaryLinesContents">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>0</y>
                  <width>797</width>
                  <height>559</height>
                 </rect>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout">
                 <item>
                  <widget class="QLabelMouseTracker" name="lblCompBoundary" native="true">
                   <property name="toolTip">
                    <string>The boundary lines separating the compartments</string>
                   </property>
                   <property name="statusTip">
                    <string>The boundary lines separating the compartments</string>
                   </property>
                   <property name="whatsThis">
                    <string>&lt;h4&gt;Compartment Geometry: Boundaries&lt;/h4&gt;
&lt;p&gt;The boundary lines separate the compartments. Each line consists of a set of vertices connected by straight lines, and these vertices will also be included in the final mesh.&lt;/p&gt;</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QLabel" name="lblMaxBoundaryPointsLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Max points:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="lblBoundaryIndexLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Boundary:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="3" colspan="2">
              <widget class="QSpinBox" name="spinMaxBoundaryPoints">
               <property name="minimum">
                <number>2</number>
               </property>
               <property name="maximum">
                <number>999</number>
               </property>
               <property name="value">
                <number>12</number>
               </property>
              </widget>
             </item>
             <item row="1" column="6">
              <widget class="QSpinBox" name="spinBoundaryZoom">
               <property name="maximum">
                <number>10</number>
               </property>
              </widget>
             </item>
             <item row="1" column="5">
              <widget class="QLabel" name="lblBoundaryZoomLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Zoom:</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tabCompartmentMesh">
          <attribute name="title">
           <string>Mesh</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="1" column="0">
              <widget class="QLabel" name="lblMaxTriangleAreaLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Max triangle area:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="4">
              <widget class="QSpinBox" name="spinMeshZoom">
               <property name="maximum">
                <number>10</number>
               </property>
              </widget>
             </item>
             <item row="1" column="1" colspan="2">
              <widget class="QSpinBox" name="spinMaxTriangleArea">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>9999</number>
               </property>
               <property name="value">
                <number>32</number>
               </property>
              </widget>
             </item>
             <item row="1" column="3">
              <widget class="QLabel" name="lblMeshZoomLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Zoom:</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="0" column="0" colspan="5">
              <widget class="QScrollArea" name="scrollMesh">
               <property name="widgetResizable">
                <bool>true</bool>
               </property>
               <widget class="QWidget" name="scrollMeshContents">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>0</y>
                  <width>797</width>
                  <height>559</height>
                 </rect>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_4">
                 <item>
                  <widget class="QLabelMouseTracker" name="lblCompMesh" native="true">
                   <property name="toolTip">
                    <string>The mesh representing the geometry of the compartment</string>
                   </property>
                   <property name="statusTip">
                    <string>The mesh representing the geometry of the compartment</string>
                   </property>
                   <property name="whatsThis">
                    <string>&lt;h4&gt;Compartment Geometry: Mesh&lt;/h4&gt;
&lt;p&gt;The compartment geometry mesh is generated from all the pixels in the compartment geometry image on the left that have the colour assigned to this compartment&lt;/p&gt;
&lt;p&gt;To change the geometry, click 'Select compartment geometry...', then click on the geometry image on the left.&lt;/p&gt;</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QLabelMouseTracker</class>
   <extends>QWidget</extends>
   <header>qlabelmousetracker.hpp</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>listCompartments</tabstop>
  <tabstop>btnAddCompartment</tabstop>
  <tabstop>btnRemoveCompartment</tabstop>
  <tabstop>listMembranes</tabstop>
  <tabstop>txtCompartmentName</tabstop>
  <tabstop>tabCompartmentGeometry</tabstop>
  <tabstop>scrollBoundaryLines</tabstop>
  <tabstop>spinBoundaryIndex</tabstop>
  <tabstop>spinMaxBoundaryPoints</tabstop>
  <tabstop>spinBoundaryZoom</tabstop>
  <tabstop>btnChangeCompartment</tabstop>
  <tabstop>spinMaxTriangleArea</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
